src/robot.py

```diff
@@ -1,30 +1,123 @@
 
-        self.driveTrain = wpilib.drive.DifferentialDrive(self.portMotorLeader, self.starboardMotorOne)
+        self.spinPID = self.spinMotor.getPIDController()
+       # self.drivePID = self.driveMotor.getPIDController()
 
         self.joy = wpilib.Joystick(0)
 
-    # Initalizes joystick to control the drivetrain
+    def teleopInit(self):
+        #Set the current position of the wheel as 0 on 
+        #the encoders when the robot is enabled
+        self.spinEncoder.setPosition(0)
+        #self.driveEncoder.setPosition(0)
+
+        #PID values (Needs heavy tuning)
+        self.spinPID.setP(0.02)
+        self.spinPID.setD(0.5)
+        self.spinPID.setFF(0)
+        self.spinPID.setI(1e-4)
+        self.spinPID.setSmartMotionMaxVelocity(0.3)
+
+        #self.drivePID.setP(0.2)
+        #self.drivePID.setD(0.3)
+        #self.drivePID.setFF(0)
+        #self.drivePID.setI(1e-4)
+        #self.drivePID.setSmartMotionMaxVelocity(0.3)
+
     def teleopPeriodic(self):
-        self.driveTrain.arcadeDrive(self.joy.getRawAxis(1), -self.joy.getRawAxis(2))
+        #METHODS OF SPINNING
+
+        #Spin when joystick is turned
+        #self.spinMotor.set(self.joy.getRawAxis(2)/20)
+
+        #Set position to joystick turn value
+        #self.spinPID.setReference(self.joy.getRawAxis(2)*5, rev.ControlType.kPosition, 0)   
+
+        #Turn via buttons (Fast and slower speeds) and stop on trigger press
+        #Used for debugging when moving the stick does something other than turn the 
+        #wheel vie the other two above methods. Trigger stop is especially useful when
+        #the motor goes out of control for some reason (saves a lot of robot restarts via e-stop)
+        if (self.joy.getRawButton(5) == True):
+            self.spinMotor.set(0.001)
+        if (self.joy.getRawButton(6) == True):
+            self.spinMotor.set(-0.2)
+
+        if (self.joy.getRawButton(3) == True):
+            self.spinMotor.set(0.1)
+        if (self.joy.getRawButton(4) == True):
+            self.spinMotor.set(-0.1)
+        
+        if (self.joy.getRawButton(1) == True):
+            self.spinMotor.set(0)
+
+        #Definitions for the current angle of the joystick. Due to how the joystick values
+        #translate into angels, I need to take this slightly differently depending on 
+        #whether the joystick is currently on the left or right side. The value returned is
+        #in degrees, with straight up as 0 and increasing clockwise to 360. (so straight right
+        #is 90, down is 180, left is 270, and technicly up is also 360 although the code
+        #deafults to 0)
+        if (self.joy.getRawAxis(1) < 0):
+            joyVectorAngle = vectorMath(-self.joy.getRawAxis(1),
+                                        self.joy.getRawAxis(0), 0)
+        elif (self.joy.getRawAxis(1) > 0):
+            joyVectorAngle = vectorMath(-self.joy.getRawAxis(1),
+                                        self.joy.getRawAxis(0), 1)
```

Comment:
I think the bug you were experiencing earlier is here, this vector maps from (0,1) to (-1,1) I think it should be backwards, try

```python
joyVectorAngle = vectorMath(-self.joy.getRawAxis(1), self.joy.getRawAxis(0), 1)
```